core/controlplane/config/stack_config.go

```diff
@@ -128,35 +55,11 @@ func (c *StackConfig) EtcdSnapshotsS3PrefixRef() (string, error) {
 	return s3path, nil
 }
 
-func (c *StackConfig) ValidateUserData() error {
-	err := userdatavalidation.Execute([]userdatavalidation.Entry{
-		{Name: "UserDataWorker", Content: c.UserDataWorker},
-		{Name: "UserDataController", Content: c.UserDataController},
-		{Name: "UserDataEtcd", Content: c.UserDataEtcd},
-	})
-
-	return err
-}
```

Comment:
Because template rendering is now lazy, validation is happening at render time inside `.Template()` function